### PR TITLE
Validation error messages correctness

### PR DESCRIPTION
**Pull request by K-Phoen**
This PR follows up on #185 with a few more validation test cases, and more importantly some fixes in the error messages Thema generates.

We focused on improving four main areas:

* CUE types should be human readable. `int32` instead of `int & >=-2147483648 & <=2147483647`
* "data" and "schema" sections of the error shouldn't be mixed up (see #140)
* references to lines and files should be correct
* missing fields should be reported as such (**note:** depending on how we see this, it's either a bugfix or a BC break)

The `mungeValidateErr()` function is still the beast that it was: we haven't found a clean way to rewrite it yet.

### CODE REVIEW

**Comment by sdboyer** [comment]
> missing fields should be reported as such (note: depending on how we see this, it's either a bugfix or a BC break)

say more? there's definitely some issues we need to work on in the bc checker around missing fields, but as i think of it, all bc checking is a lineage-bind-time concern - it should all be decided by the time we get to data validation

**Review by sdboyer — CHANGES_REQUESTED** [review]
This looks _awesome_. Well done! Would just like to see one thing documented a bit more before we merge

File: schema.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func (sch *schemaDef) Validate(data cue.Value) (*Instance, error) {
 	// }
 
 	x := sch.def.Unify(data)
-	if err := x.Validate(cue.Final(), cue.All()); err != nil {
+	if err := x.Validate(cue.Concrete(true), cue.All()); err != nil {
```
[the remark this answers is not in the document]
Comment (K-Phoen) [reply schema.go]: I pushed a commit to address this, can you have a look? :)

[the remark this answers is not in the document]
Comment (joanlopez) [reply schema.go]: I'm wondering now, could this cause any errors with intermediate "optional" fields? Like, what happens if we have something like `joinSchema` that's not populated at all? 🤔 @sdboyer

[the remark this answers is not in the document]
Comment (sdboyer) [reply schema.go]: Adjustments to the comments look good!

**Comment by sdboyer** [comment]
I'm going to test and make sure that that dashboard validation still works before approving (looking forward to having more automated tests for this!)

**Comment by sdboyer** [comment]
looking good on a dashboard - in it goes!

**Review by sdboyer — APPROVED** [review]

**Pull request merged**